R/cube-result.R

```diff
@@ -388,6 +411,122 @@ setMethod("margin.table", "CrunchCube", function (x, margin=NULL) {
     cubeMarginTable(x, margin)
 })
 
+#' @rdname cube-computing
+#' @export
+setMethod("collapse.dimensions", "CrunchCube", function (x, margin=NULL) {
+    # ensure that we have sensible margin input
+    selecteds <- is.selectedDimension(x@dims)
+    check_margins(margin, selecteds)
+    
+    # translate from user-cube margins to real-cube margins and establish the
+    # two groups of margins: those to collapse and those to keep
+    margins_to_collapse <- user2real(margin, cube = x)
+    margins_to_keep <- setdiff(seq_along(x@dims), margins_to_collapse)
+    
+    # if there are any _items in the margin to collapse, be wary!
+    collapsed_items <- endsWith(getDimTypes(x)[margins_to_collapse], "_items")
+
+    # grab the old cube structure and over-write it with new subset data.
+    out <- x
+    
+    # iterate through measures, collapsing the dimension(s) specified.
+    out@arrays[] <- lapply(out@arrays, function(arr){
```

Comment:
This is a pretty big function to stick in a `lapply` and I found it a bit hard to follow. It might be easier to read if it were pulled into its own function, and purified (make the scoped arguments explicit arguments). That way you could pull out some of the motivation comments into function documentation.